src/java/org/apache/cassandra/net/Message.java

```diff
@@ -889,18 +889,19 @@ private void serializeHeaderPre40(Header header, DataOutputPlus out, int version
             serializeParams(addFlagsToLegacyParams(header.params, header.flags), out, version);
         }
 
-        private Header deserializeHeaderPre40(DataInputPlus in, int version) throws IOException
+        private Header deserializeHeaderPre40(DataInputPlus in, InetAddressAndPort peer, int version) throws IOException
         {
             validateLegacyProtocolMagic(in.readInt());
             int id = in.readInt();
             long currentTimeNanos = approxTime.now();
             MonotonicClockTranslation timeSnapshot = approxTime.translate();
             long creationTimeNanos = calculateCreationTimeNanos(in.readInt(), timeSnapshot, currentTimeNanos);
-            InetAddressAndPort from = inetAddressAndPortSerializer.deserialize(in, version);
+            // skip from field
+            inetAddressAndPortSerializer.deserialize(in, version);
```

Comment:
is this change needed? I traced caller up to the following method and it's not used.
```
        public <T> Message<T> deserialize(DataInputPlus in, InetAddressAndPort peer, int version) throws IOException
        {
            return version >= VERSION_40 ? deserializePost40(in, peer, version) : deserializePre40(in, peer, version);
        }
```

Reply:
It is used when CC node talks to a DSE node. In this case the messaging protocol is downgraded to 100 (C* 3.x).

Reply:
I couldn't find it either. The other one (https://github.com/datastax/cassandra/pull/904/files#diff-508770b4fd2a653672a8cc604fd6e843ea8464aa931da7982047c71f07470efeR941) is used for cross-version messaging.
I must have changed this method for consistency.

Reply:
Given it's not used, it should be safe..